src/main/java/ch/uzh/ifi/hase/soprafs21/objects/GameEngine.java

```diff
@@ -74,6 +74,7 @@ public void addUserToWaitingRoom(User user){
                     String userIdentity = userInWaitingRoom.getSessionIdentity();
                     webSocketService.sendGameAssignmentMessageToWaitingRoom(userIdentity, createdGame.getPlayers(), createdGame.getGameId());
                 }
+                waitingRoom.resetUsers();
```

Comment:
does this not already happening in waitingRoom.getXNumberOfUsers() with the line: User newUserToAddToGamesession = userQueue.remove(0); ? maybe we should better decrease the usercount directly in here after this line, or even better make a getUserCount() which actually computes the usercount every time from the current queue